specs/block_spec.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'spec_helper'
+
+describe "Block class" do
+  before do
+    Availability.create_calendar
+  end
+
+  describe "initialize new block" do
+    it "returns an error if you try and book a block less than 2 rooms or more than 5 rooms" do
+      proc { Block.new(1, 2017, 9, 28, 2017, 9, 29, "Smith")}.must_raise ArgumentError
+      proc { Block.new(9, 2017, 9, 28, 2017, 9, 29, "Jones")}.must_raise ArgumentError
+    end
+
+    it "creates a new instance of block class" do
+      block = Block.new(4, 2017, 9, 28, 2017, 9, 29, "Johnson")
+      block.must_be_instance_of Block
+    end
+
+    it "returns a block id for administrator reference" do
+      block = Block.new(4, 2017, 9, 28, 2017, 9, 29, "Hayward")
+      block.block_id.must_equal "Hayward"
+    end
+
+    it "raises an error if you choose a block ID that has already been taken" do
+      Block.new(4, 2017, 9, 28, 2017, 9, 29, "Brennan")
+      proc { Block.new(3, 2017, 9, 28, 2017, 9, 29, "Brennan")}.must_raise ArgumentError
+    end
+
+    it "returns an array of rooms that have been blocked" do
+      block = Block.new(4, 2017, 9, 28, 2017, 9, 29, "Cooper")
+      block.blocked_rooms.must_be_instance_of Array
+      block.blocked_rooms.count.must_equal 4
+    end
+
+    it "has a single check-in and check-out date that applies to all rooms in the block" do
```

Comment:
You need to check edge cases, like invalid days, repeated reservations for the same room, reserving all the rooms etc.